--- cygwinccompiler.py	2020-12-21 18:06:58.000000000 +0900
+++ cygwinccompiler_after.py	2021-01-30 14:03:47.499694848 +0900
@@ -82,6 +82,24 @@
         elif msc_ver == '1600':
             # VS2010 / MSVC 10.0
             return ['msvcr100']
+        elif msc_ver == '1700':
+            # Visual Studio 2012 / Visual C++ 11.0
+            return ['msvcr110']
+        elif msc_ver == '1800':
+            # Visual Studio 2013 / Visual C++ 12.0
+            return ['msvcr120']
+        elif msc_ver == '1900':
+            # Visual Studio 2015 / Visual C++ 14.0
+            # "msvcr140.dll no longer exists"
+            return ['vcruntime140']
+        elif msc_ver == '1916':
+            # Visual Studio 2015 / Visual C++ 14.0
+            # "msvcr140.dll no longer exists"
+            return ['vcruntime140']
+        elif msc_ver == '1928':
+            # Visual Studio 2015 / Visual C++ 14.0
+            # "msvcr140.dll no longer exists"
+            return ['vcruntime140']
         else:
             raise ValueError("Unknown MS Compiler version %s " % msc_ver)
 
